internal/pkg/cli/svc_deploy.go

```diff
@@ -216,6 +217,9 @@ func (o *deploySvcOpts) Execute() error {
 	if err != nil {
 		return err
 	}
+	if o.forceNewUpdate && o.svcType == manifestinfo.StaticSiteType {
```

Comment:
I was searching up where is `o.svcType` figured out, and realized it was in `configureClients`, which is quite unobvious 😓 What do you think of just moving the following lines from `configureClients` to here in `Execute()`:
```go
svc, err := o.store.GetService(o.appName, o.name)
if err != nil {
  return fmt.Errorf("get service %s configuration: %w", o.name, err)
}
o.svcType = svc.Type
```

Reply:
We should also mention this in our flag description!

Reply:
> We should also mention this in our flag description!

Sounds good! I'll mention it in the description

> What do you think of just moving the following lines from configureClients to here in Execute()

I feel like the same logic could be applied to the other logic in `configureClients` too. The purpose of that function is for facilitating the unit test so that the test can focus on the main logic for execute rather than setting up prerequisites. If that is the case, maybe it's better to keep it as is 🤔 Maybe it would be less confusing if we cache the `svc` as a concrete struct like `env` as opposed to `svcType`?

Reply:
My main problem with it is that `svcType` is not a client, so it is a pretty hidden secret that `o.svcType` is figured out in `configureClients`.

It is a piece of information returned by `o.store`, which _is_ a client and can be mocked. I know you replied in the other thread that `GetService` is not a main logic to be tested. However, it _is_ a prerequisite for the rest of deployment, and has to be invoked in for the whole `Execute` to complete. My argument is that, if we remove that `o.store.GetService` line from `svc_deploy.go`, our unit test won't catch that, but our `svc deploy` will certainly break.

So I argue that it _is_ a main piece of logic, however, it could be moved to `Ask`.

Reply:
Ok moved the logic to `Ask`